ecadlabs/taqueria
# Fixes #524, initial version of jest plugin

The is the initial implementation of the jest plugin.

This PR does not include:
- e2e tests
- adjustments to the Github workflow to build and publish the NPM package
- test stubs
- test templates

The initial implementation exposes a new task called `test`.

### Usage:

> Create the default partition
1) Run: `taq test --init`
2) A _tests_ directory with a jest.config.js file should be created
3) The _tests/jest.config.js_ should be overriding the newly created _.taq/jest.config.js_ file

> To run some tests using Jest
1) Create _tests/example.spec.ts_  with the following contents:
```
describe('Jest example', () => {
    test('Example of an assertion', () => {
        expect("test").toEqual("test")
    })
})
```
2) Run: `taq test`

> To run a particular test
1) Create _tests/example.spec.ts_  with the following contents:
```
describe('Jest example', () => {
    test('Example of an assertion', () => {
        expect("test").toEqual("test")
    })
})
```
2) Create _tests/foobar.spec.ts_ with the following contents:
```
describe('Jest example of failure', () => {
    test('Example of an falied assertion', () => {
        expect("test").toEqual("foobar")
    })
})
```
3) Run `taq test -t foobar.spec.ts`
